server/controllers/files.coffee

```diff
@@ -132,124 +132,136 @@ module.exports.create = (req, res, next) ->
             part.on 'data', (buffer) ->
                 fields[part.name] = buffer.toString()
 
-            return
-
-        # We assume that only one file is sent.
-        # we do not write a subfunction because it seems to load the whole
-        # stream in memory.
-        name = fields.name
-        path = fields.path
-        overwrite = fields.overwrite
-
-        # we have no name for this file, give up
-        if not name or name is ""
-            err = new Error "Invalid arguments: no name given"
-            err.status = 400
-            return next err
-
-        upload = true
-
-        # while this upload is processing
-        # we send usage.application to prevent auto-stop
-        # and we defer parents lastModification update
-        keepAlive = ->
-            if upload
-                feed.publish 'usage.application', 'files'
-                setTimeout keepAlive, 60*1000
-                resetTimeout()
-
-        # if anything happens after the file is created
-        # we need to destroy it
-        rollback = (file, err) ->
-            file.destroy (delerr) ->
-                # nothing more we can do with delerr
-                log.error delerr if delerr
-                if isStorageError err
-                    res.send
-                        error: true
-                        code: 'ESTORAGE'
-                        msg: "modal error size"
-                    , 400
-                else
-                    next err
-
-        attachBinary = (file) ->
-            # request-json requires a path field to be set
-            # before uploading
-            part.path = file.name
-            metadata = name: "file"
-            file.attachBinary part, metadata, (err) ->
-                upload = false
-                return rollback file, err if err
-
-                # index the file in cozy-indexer for fast search
-                file.index ["name"], (err) ->
-                    # we ignore indexing errors
-                    log.debug err if err
-                    # send email or notification of file changed
-                    who = req.guestEmail or 'owner'
-                    sharing.notifyChanges who, file, (err) ->
-                        # we ignore notification errors
-                        log.debug err if err
-                        res.send file, 200
-
-        now = moment().toISOString()
-
-        # Check that the file doesn't exist yet.
-        path = normalizePath path
-        fullPath = "#{path}/#{name}"
-        File.byFullPath key: fullPath, (err, sameFiles) =>
-            return next err if err
+        else
```

Comment:
Yes, but it's a tough work. Because any wrong assignation can lead to the full loading of the file in memory.